### DISCUSSION TITLE
apt update error: tzdata

### DISCUSSION THREAD

**Discussion by cociweb** [docker-mailserver/docker-mailserver#2801] [Q&A] [answered]
Hello,

I've tried to update the apt packages inside the docker image and the tzdata packages continously gives an error. I've recreated the container and the issue is the same. Maybe I do something wrong, with the timezone data and/or env variables or it's a general issue with the container?

```
apt upgrade

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
The following packages will be upgraded:
  tzdata
1 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Need to get 284 kB of archives.
After this operation, 6144 B disk space will be freed.
Do you want to continue? [Y/n] y
Get:1 http://deb.debian.org/debian bullseye/main amd64 tzdata all 2021a-1+deb11u5 [284 kB]
Fetched 284 kB in 0s (956 kB/s)
Preconfiguring packages ...
(Reading database ... 18734 files and directories currently installed.)
Preparing to unpack .../tzdata_2021a-1+deb11u5_all.deb ...
Unpacking tzdata (2021a-1+deb11u5) over (2021a-1+deb11u4) ...
dpkg: error processing archive /var/cache/apt/archives/tzdata_2021a-1+deb11u5_all.deb (--unpack):
 unable to make backup link of './usr/share/zoneinfo/Etc/UTC' before installing new version: Invalid cross-device link
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Errors were encountered while processing:
 /var/cache/apt/archives/tzdata_2021a-1+deb11u5_all.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

I've checked the above mentioned /var/cache/... path, but nothing there. Finally the tzdata package is not upgraded.
/etc/localtime is mounted as /etc/localtime:/etc/localtime:ro from the ubuntu 22.04 host machine.

Do you have any hint where should I start?

**Comment by casperklein** [comment, marked as answer]
> Maybe I do something wrong, with the timezone data and or env variables or it's a general issue with the container?

The latter. I can confirm this error and created an [issue](https://github.com/docker-mailserver/docker-mailserver/issues/2802) for that.

**Reply by cociweb** [reply]
Great! Thank's in this case, let's close this discussion, and let's continue it under the issue!
